apps/mobile/src/components/ChatDetailScreen/components/MessagesList.tsx

```diff
@@ -15,16 +25,29 @@ function renderItem({
 }
 
 function MessagesList(): JSX.Element {
-  const {messagesListAtomAtoms} = useMolecule(chatMolecule)
+  const {
+    messagesListAtomAtoms,
+    handleIsRevealIdentityOrContactRevealMessageVisibleActionAtom,
+  } = useMolecule(chatMolecule)
   const dataAtoms = useAtomValue(messagesListAtomAtoms)
+  const handleIsRevealIdentityOrContactRevealMessageVisible = useSetAtom(
+    handleIsRevealIdentityOrContactRevealMessageVisibleActionAtom
+  )
 
   return (
     <FlashList
       data={dataAtoms}
       estimatedItemSize={54}
+      contentContainerStyle={contentStyle.contentContainerStyle}
       keyExtractor={atomKeyExtractor}
       inverted
       renderItem={renderItem}
+      onViewableItemsChanged={
+        handleIsRevealIdentityOrContactRevealMessageVisible
```

Comment:
Debounce - when user has a lot of messages + user has lot of contacts there might be a big performance impact